### PR TITLE
🔧(project) add indexer swarm cronjob with documentation

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

Warren-TdBP needs to know relations between activities in a course.

## Proposal

Adding a swarm cronjob that executes an indexation of all courses and their contents daily, as an example.
Adding instructions to configure Moodle web services and to deploy the cronjob.

### CODE REVIEW

File: production/env.d/api.env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Warren API
 WARREN_API_SERVER_PORT=8100
-WARREN_LRS_HOSTS=http://ralph:8080
+WARREN_LRS_HOSTS=http://ralph-app-1:8080
```
Comment (lunika) [inline production/env.d/api.env:3]: Is it really needed ? This is the name exposed by `docker` but not `docker compose` I guess

Comment (wilbrdt) [reply production/env.d/api.env:3]: For a demo purpose, Ralph is started from its project and both Ralph and Warren app are called `app` in their resp. `docker-compose.yml`. In a real case, it should always be adapted to the Ralph URL anyway.
I would keep it this way for ease of a quick demo stack, WDYT?

File: production/README.md [hunk 2]
```diff
@@ -89,6 +89,48 @@ Ensure Warren-TdBP API is connected to a Ralph instance for fetching statements.
 using Docker Swarm to deploy Ralph, make sure both services are running on the same
 `backend` network.
 
+### Configuring the indexer cronjob
+
+To understand the activities of a course, the Experience Index stores the relationships
+between each activity within a course. We propose using a cronjob to index courses and
+their content at scheduled times. This cronjob will index all courses and content in the
```
Comment (quitterie-lcs) [inline production/README.md]:
```suggestion
their content at scheduled times. This cronjob will index all courses and contents in the
```

File: production/README.md [hunk 3]
```diff
@@ -89,6 +89,48 @@ Ensure Warren-TdBP API is connected to a Ralph instance for fetching statements.
...
+
+In order for the Experience Index to access the Moodle courses and their content, we
+need to configure a web service under `Moodle > Web services` and following the 10 steps
+provided by Moodle:
+1. Enable web service: `Yes`
+2. Enable protocols: `rest`
+3. Create a specific user: `warren_tdbp`
+4. Check its user capability
+5. Select a service: add a new service (let's name it `Warren-TdBP - XI`) 
+6. Add functions: new service `Warrem-TdBP - XI` needs to have the following functions:
```
Comment (quitterie-lcs) [inline production/README.md]:
```suggestion
6. Add functions: new service `Warren-TdBP - XI` needs to have the following functions:
```

**Review by lunika — APPROVED** [review]

**Pull request merged**